cpp/src/copying/slice.cpp

```diff
@@ -33,30 +34,27 @@ std::vector<column_view> slice(column_view const& input,
 {
   CUDF_EXPECTS(indices.size() % 2 == 0, "indices size must be even");
 
-  if (indices.empty()) { return {}; }
+  if (indices.empty()) return {};
 
   auto null_counts = cudf::detail::segmented_count_unset_bits(input.null_mask(), indices, stream);
+  auto const children = std::vector<column_view>(input.child_begin(), input.child_end());
 
-  std::vector<column_view> children{};
-  for (size_type i = 0; i < input.num_children(); i++) { children.push_back(input.child(i)); }
-
-  std::vector<column_view> result{};
-  for (size_t i = 0; i < indices.size() / 2; i++) {
+  auto op = [&](auto i) {
     auto begin = indices[2 * i];
     auto end   = indices[2 * i + 1];
     CUDF_EXPECTS(begin >= 0, "Starting index cannot be negative.");
     CUDF_EXPECTS(end >= begin, "End index cannot be smaller than the starting index.");
     CUDF_EXPECTS(end <= input.size(), "Slice range out of bounds.");
-    result.emplace_back(input.type(),
-                        end - begin,
-                        input.head(),
-                        input.null_mask(),
-                        null_counts[i],
-                        input.offset() + begin,
-                        children);
-  }
-
-  return result;
+    return column_view{input.type(),
+                       end - begin,
+                       input.head(),
+                       input.null_mask(),
+                       null_counts[i],
+                       input.offset() + begin,
+                       children};
+  };
+  auto begin = cudf::detail::make_counting_transform_iterator(0, op);
+  return {begin, begin + indices.size() / 2};
```

Comment:
I know this isn't necessary, but I feel like it will be less confusing to most readers. 
```suggestion
  return std::vector<column_view>{begin, begin + indices.size() / 2};
```